assets/src/stories-editor/components/inserter/menu.js

```diff
@@ -409,9 +408,7 @@ export default compose(
 				return true;
 			}
 
-			// canInsertBlockType() alone is not enough, see https://github.com/WordPress/gutenberg/issues/14515
-			const destinationBlockListSettings = getBlockListSettings( destinationRootClientId );
-			return canInsertBlockType( name, getCurrentPage() ) && destinationBlockListSettings && destinationBlockListSettings.allowedBlocks.includes( name );
```

Comment:
Hm, why does this old code reference `getCurrentPage()` in one instance and `destinationRootClientId` in the other? Should they both be `destinationRootClientId` and thus this should be used in the refactored code as well?

Reply:
@swissspidy it seems you [made the original implementation here](https://github.com/ampproject/amp-wp/commit/bb4a35c71fb673e77b2bcbd5167a15ae1c1dc2a2#diff-e633a19ee56c05aab9455a179f61a078R387-R395). Can you help me resolve this? Is my refactor correct or should it reference `destinationRootClientId `.

Reply:
This component an almost 1:1 copy of the InserterMenu component in `@wordpress/block-editor`. The inserter in the regular block editor is not only shown in the top toolbar, but also when using the appenders within a columns block or between paragraphs.

In our use case, the component is only used in the top toolbar though, and as such actually doesn't need the whole `destinationRootClientId` logic — the destination will always be the current page block.

So your changes look correct 👍